plenum/test/view_change/test_client_req_during_view_change.py

```diff
@@ -1,49 +1,47 @@
 import pytest
 
-from plenum.common.exceptions import RequestNackedException
-from plenum.test.helper import sdk_send_random_and_check, \
-    sdk_send_random_requests, sdk_get_and_check_replies, sdk_gen_request, \
-    checkDiscardMsg
-
-
-def test_client_msg_discard_in_view_change_integration(txnPoolNodeSet,
-                                                       looper,
-                                                       sdk_pool_handle,
-                                                       sdk_wallet_client):
-    '''
-    Check that client requests sent in view change will discard.
-    '''
-    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
-                              sdk_wallet_client, 4)
-
-    for node in txnPoolNodeSet:
-        node.view_changer.view_change_in_progress = True
-    discard_reqs = sdk_send_random_requests(looper, sdk_pool_handle,
-                                            sdk_wallet_client, 1)
-    with pytest.raises(RequestNackedException) as e:
-        sdk_get_and_check_replies(looper, discard_reqs)
-        assert "Client request is discarded since view " \
-               "change is in progress" in e.args[0]
-
-
-def test_client_msg_discard_in_view_change_with_dict(txnPoolNodeSet):
-    node = txnPoolNodeSet[0]
-    node.view_changer.view_change_in_progress = True
-    node.send_nack_to_client = check_nack_msg
-
-    msg = sdk_gen_request("op").as_dict
-    node.unpackClientMsg(msg, "frm")
-    checkDiscardMsg([node, ], msg, "view change in progress")
-
-
-def test_client_msg_discard_in_view_change_with_request(txnPoolNodeSet):
-    node = txnPoolNodeSet[0]
-    node.view_changer.view_change_in_progress = True
-    node.send_nack_to_client = check_nack_msg
-
-    msg = sdk_gen_request("op")
-    node.unpackClientMsg(msg, "frm")
-    checkDiscardMsg([node, ], msg.as_dict, "view change in progress")
+from plenum.common.constants import NODE, TXN_TYPE, GET_TXN
+from plenum.test.helper import sdk_gen_request, checkDiscardMsg
+from plenum.test.testing_utils import FakeSomething
+
+
+@pytest.fixture(scope='function')
+def test_node(test_node):
+    test_node.view_changer = FakeSomething(view_change_in_progress=True,
+                                           view_no=1)
+    return test_node
+
+
+def test_client_write_request_discard_in_view_change_with_dict(test_node):
+    test_node.send_nack_to_client = check_nack_msg
+
+    msg = sdk_gen_request({TXN_TYPE: NODE}).as_dict
+    test_node.unpackClientMsg(msg, "frm")
+    checkDiscardMsg([test_node, ], msg, "view change in progress")
+
+
+def test_client_get_request_not_discard_in_view_change_with_dict(test_node):
```

Comment:
It would be great to have tests for non-get-txn requests, as well as for actions.